api/valor_api/backend/core/label.py

```diff
@@ -26,71 +26,95 @@ def validate_matching_label_keys(
     ----------
     db : Session
         The database Session to query against.
-    prediction_filter : schemas.Filter
-        The filter to be used to query predictions.
-    groundtruth_filter : schemas.Filter
-        The filter to be used to query groundtruths.
+    dataset_names : list[str]
+        The list of required datasets by name.
+    model_name : str
+        The required model by name.
     label_map: LabelMapType, optional
         Optional mapping of individual labels to a grouper label. Useful when you need to evaluate performance using labels that differ across datasets and models.
 
-
     Raises
     -------
     ValueError
         If the distinct ground truth label keys don't match the distinct prediction label keys for any datum.
     """
 
-    gts = generate_select(
-        models.Annotation.datum_id.label("datum_id"),
-        models.Label.key.label("label_key"),
-        models.Label.value.label("label_value"),
-        filters=groundtruth_filter,
-        label_source=models.GroundTruth,
-    ).alias()
-
-    gt_label_keys_by_datum = (
+    gt_labels_by_datum = (
         select(
-            gts.c.datum_id,
-            func.array_agg(gts.c.label_key + ", " + gts.c.label_value).label(
+            models.Datum.id.label("datum_id"),
+            func.array_agg(models.Label.key + ", " + models.Label.value).label(
                 "gt_labels"
             ),
         )
-        .select_from(gts)
-        .group_by(gts.c.datum_id)
+        .select_from(models.Datum)
+        .join(
+            models.Dataset,
+            and_(
+                models.Dataset.id == models.Datum.dataset_id,
+                models.Dataset.name.in_(dataset_names),
```

Comment:
could the filters be an important addition here? is there an edge case where selecting all labels from a dataset/model wouldn't pass this validation, but filtering the labels using some user's filters would pass this validation?

Reply:
Ill revert to using `from valor_api.backend.metrics.metric_utils import prepare_filter_for_evaluation` as it properly formats the filter for evaluation.

Reply:
thank you!